### DISCUSSION TITLE
Hi there, where can i find the `.LIB` file of the 'av_libglesv2.dll'?

### DISCUSSION THREAD

**Discussion by jcyuan** [AvaloniaUI/Avalonia#13990] [Q&A] [answered]
i believe Avalonia official built this dll themselives.
from `https://github.com/AvaloniaUI/Avalonia/pull/10204`, so @maxkatz6 , man you did it right? could you please help to provide the lib file of v2.1.22045 dll please? 
and is it possible to provide it and those header files with the dll file together in the nuget package please?
thank you so much!

**Comment by jcyuan** [comment]
why i need it is because i have another project uses ANGLE too, but normally the ANGLE dll name is libGLESv2.dll, not sure why Ava has a different name with it, anyway i can change my project to use Ava's dll easily, this way i can remove the redundant dlls to make it more neat.

**Comment by maxkatz6** [comment]
Use dllimport redirection
https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.nativelibrary.setdllimportresolver?view=net-8.0

**Comment by jcyuan** [comment]
> Use dllimport redirection https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.nativelibrary.setdllimportresolver?view=net-8.0

no sir, that project is a c++ project, load dll is too hard to use, my purpose is just use the lib in my linker to output my dll from my project, so it will use av_libGLESv2.dll instead of the ANGLE official named dll.
because there is no way to modify the dll name in ava unless modify the source code.
or could you please tell me where can i find this version of source from ANGLE to compile please, i can do that myself too.
thank you.

**Reply by maxkatz6** [reply]
You can redirect "av_libGLESv2.dll" used by Avalonia to "angle.dll" or whatever name you wish of a single dll.

**Comment by jcyuan** [comment]
@maxkatz6 Hi Max,
i just found that this Resolver only works for P/Invoke.
and NativeLibrary.Load won't use this resolver to parse the dependencies which means this way not work for me...

**Comment by kekekeks** [comment]
https://www.asawicki.info/news_1420_generating_lib_file_for_dll_library

**Reply by maxkatz6** [reply, marked as answer]
Angle.lib was already built for Avalonia previously for AOT: https://github.com/AvaloniaUI/Avalonia/issues/9503#issuecomment-1824425574 cc @jcyuan

**Comment by jcyuan** [comment]
> https://www.asawicki.info/news_1420_generating_lib_file_for_dll_library

well man, i actaully know this way, just you have to make that def first (maybe by dumpbin or manually write it), and you have to check errors etc which might make you feel bad. buy anyway thanks~

@maxkatz6 thanks so much~ got it, lovely~ 🙏
